src/main/groovy/io/seqera/wave/service/builder/ContainerBuildServiceImpl.groovy

```diff
@@ -161,6 +161,44 @@ class ContainerBuildServiceImpl implements ContainerBuildService {
         }
     }
 
+    //this method adds labels in singularity container
+    protected static String addLabels(String containerFile, BuildRequest req){
```

Comment:
I would change the signature of the method to `String containerFile, Map<String, String> labels` , it is clearer.  
Then the conditional in the line below would become `if(labels)`